### PR TITLE
Fix issues with gradle project generation

### LINKED ISSUE #4365: Gradle build doesn't work if the plugin isn't in the local repository
**Describe the bug**
When running any Gradle task, the build fails 

**Expected behavior**
Gradle build should work

**Actual behavior**
It doesn't work. The following stacktrace is shown: 

```bash
➜  code-with-quarkus ./gradlew quarkusDev

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'code-with-quarkus'.
> Could not resolve all artifacts for configuration ':classpath'.
   > Could not find io.quarkus:quarkus-gradle-plugin:0.23.2.
     Searched in the following locations:
       - file:/home/ggastald/.m2/repository/io/quarkus/quarkus-gradle-plugin/0.23.2/quarkus-gradle-plugin-0.23.2.pom
       - file:/home/ggastald/.m2/repository/io/quarkus/quarkus-gradle-plugin/0.23.2/quarkus-gradle-plugin-0.23.2.jar
     Required by:
         project :

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 404ms
```

**To Reproduce**
1. Run code.quarkus.io with this PR: https://github.com/quarkusio/code.quarkus.io/pull/163  and generate a Gradle project
2. Unzip the generated project and run `./gradlew quarkusDev`

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
```
Linux ggastald-laptop 5.2.15-200.fc30.x86_64 #1 SMP Mon Sep 16 15:17:36 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```
 - Output of `java -version`: 
```
openjdk version "1.8.0_212"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_212-b03)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.212-b03, mixed mode)
```
 - GraalVM version (if different from Java): Not applicable
 - Quarkus version or git rev: 0.23.2

Additional Context: 

I don't have the quarkus-gradle-plugin in my local repository

### PR DESCRIPTION
**Pull request by geoand**
This had an effect of removing for example the Kotlin plugin when kotlin was being used

Fixes #4365

### CODE REVIEW

File: devtools/common/src/main/java/io/quarkus/cli/commands/file/BuildFile.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
     protected final static Printer PRINTER = new Printer();
 
-    private final ProjectWriter writer;
+    protected final ProjectWriter writer;
```
Comment (maxandersen) [inline devtools/common/src/main/java/io/quarkus/cli/commands/file/BuildFile.java]: there is a `getWriter()` method - use that instead.

Comment (geoand) [reply devtools/common/src/main/java/io/quarkus/cli/commands/file/BuildFile.java]: Didn't see it, good catch!

File: devtools/common/src/main/java/io/quarkus/cli/commands/file/GradleBuildFile.java [hunk 2]
```diff
@@ -30,18 +30,6 @@
 
     public GradleBuildFile(ProjectWriter writer) throws IOException {
         super(writer, BuildTool.GRADLE);
-        if (writer.exists(SETTINGS_GRADLE_PATH)) {
```
Comment (maxandersen) [inline devtools/common/src/main/java/io/quarkus/cli/commands/file/GradleBuildFile.java]: MavenBuildFile has similar forced initialization - should we remove that too in constructor ?

Comment (geoand) [reply devtools/common/src/main/java/io/quarkus/cli/commands/file/GradleBuildFile.java]: That would probably make sense. Let me try it and see what happens

Comment (geoand) [reply devtools/common/src/main/java/io/quarkus/cli/commands/file/GradleBuildFile.java]: Works!

File: devtools/common/src/main/java/io/quarkus/cli/commands/file/GradleBuildFile.java [hunk 3]
```diff
@@ -55,9 +43,26 @@ public void close() throws IOException {
...
     @Override
     public void completeFile(String groupId, String artifactId, String version) throws IOException {
+        init();
         completeSettingsContent(artifactId);
         completeBuildContent(groupId, version);
-        completeProperties(artifactId);
+        completeProperties();
+    }
+
+    // this is only visible for testing purposes
```
Comment (maxandersen) [inline devtools/common/src/main/java/io/quarkus/cli/commands/file/GradleBuildFile.java]: did you forget to commit some tests ? :)

Comment (geoand) [reply devtools/common/src/main/java/io/quarkus/cli/commands/file/GradleBuildFile.java]: Nope, I forgot to remove this comment 🤪. I'm out for a short while, will update shortly

Comment (geoand) [reply devtools/common/src/main/java/io/quarkus/cli/commands/file/GradleBuildFile.java]: Fixed

**Comment by geoand** [comment]
@maxandersen OK to merge this?

**Comment by gastaldi** [comment]
I can confirm that this PR fixes 4365

**Comment by geoand** [comment]
PR updated so now the both Java and Kotlin project generation should work

**Review by gastaldi — APPROVED** [review]

**Review by aloubyansky — APPROVED** [review]

**Pull request merged**